packages/canary-web-components/src/components/ic-time-selector/ic-time-selector.stories.js

```diff
@@ -0,0 +1,185 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import { html } from "lit-html";
+import readme from "./readme.md";
+
+export default {
+  title: "Web Components/Time Selector",
+  component: "ic-time-selector",
+  parameters: {
+    componentAPI: {
+      data: readme,
+    },
+  },
+};
+
+/**
+ * Use the time selector component to manually select a time in different formats.
+ *
+ * Click the 'Component API' tab to view all the available attributes, events and slots for time selector.
+ *
+ * To use the time selector component, import `@ukic/canary-web-components` into your application.
+ *
+ * Times can be selected by scrolling through each column or using the up and down arrow keys.
+ *
+ * To move between hour, minute and second using the keyboard, provide the required values based on the placeholders or use `:`.
+ *
+ */
+export const Default = {
+  render: () => html`<ic-time-selector></ic-time-selector>`,
+  name: "Default",
+};
+
+/**
+ * Retrieving the time selector value via `icChange` returns the time as a Date object.
+ *
+ * The event returns the Date object once hour, minute and second have been selected by clicking the tick icon button.
+ */
+export const IcChangeEvent = {
```

Comment:
Clearing the selected value using the clear button doesn't cause an icChange event to be emitted - I feel like it should? 

The date picker emits an icChange event when you click the "Clear" on the dropdown

Reply:
should it also return focus to the HH selector? as currently the focus gets lost completely (because the clear button becomes disabled)